cloudesire-api-client-dto/src/main/java/com/cloudesire/platform/apiclient/dto/model/dto/UserPatchDTO.java

```diff
@@ -0,0 +1,64 @@
+package com.cloudesire.platform.apiclient.dto.model.dto;
+
+public class UserPatchDTO extends DTO
+{
+    private Action action;
+
+    private String provider;
+
+    private Object credentials;
+
+    public UserPatchDTO( Action action )
+    {
+        this.action = action;
+    }
+
+    public UserPatchDTO()
+    {
+    }
+
+    public Action getAction()
+    {
+        return action;
+    }
+
+    public void setAction( Action action )
+    {
+        this.action = action;
+    }
+
+    public String getProvider()
+    {
+        return provider;
+    }
+
+    public void setProvider( String provider )
+    {
+        this.provider = provider;
+    }
+
+    public Object getCredentials()
+    {
+        return credentials;
+    }
+
+    public void setCredentials( Object credentials )
+    {
+        this.credentials = credentials;
+    }
+
+    public enum Action
+    {
+        BLOCK,
+        UNBLOCK,
+        ACTIVATE,
+        APPROVE,
+        SAVE_CLOUD_CREDENTIALS
+    }
+
+    public static class Fields
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Add a private constructor to hide the implicit public one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.cloudesire.com/coding_rules#rule_key=squid%3AS1118)